TypeScript-Playground/src/typescript-assignment.js

```diff
@@ -219,3 +219,91 @@ export class LinkedList {
         return null;
     }
 }
+class Folder {
+    path;
+    size;
+    contents;
+    constructor(path) {
+        this.path = path;
+        this.size = 0;
+        this.contents = [];
+    }
+    getSize() {
+        /*let totalSize = 0;
+        for (const content of this.contents) {
+          if (content.path.includes(".")) {
+            totalSize += content.size;
+          } else {
+            totalSize += content.getSize();
+          }
+        } */
+        const totalSize = this.contents.reduce((acc, content) => content.path.includes(".")
+            ? acc + content.size
+            : acc + content.getSize(), 0);
+        return totalSize;
+    }
+    getPath() {
+        return this.path;
+    }
+    findFolder(path) {
+        if (this.path === path) {
+            return this;
+        }
+        else {
+            for (const content of this.contents) {
+                if (content instanceof Folder) {
+                    const found = content.findFolder(path);
+                    if (found) {
+                        return found;
+                    }
+                }
+            }
+        }
+        return undefined;
+    }
+    addFile(file) {
+        this.contents.push(file);
+    }
+    addFolder(folder) {
+        this.contents.push(folder);
+    }
+}
+export class FileSystem {
+    root;
+    constructor() {
+        this.root = new Folder("/");
+    }
+    createFile(path, size) {
+        const dividedPathNames = path.split("/");
+        const fileName = dividedPathNames.pop();
+        const parentFolderPath = dividedPathNames.join("/") || "/";
+        const parentFolder = this.root.findFolder(parentFolderPath);
+        if (!parentFolder || !fileName) {
+            return undefined;
+        }
+        const newFile = {
+            path: `${path}`,
+            size,
+            getSize: function () {
+                return this.size;
+            },
+            getPath: function () {
+                return this.path;
+            },
+        };
+        parentFolder.addFile(newFile);
+        return newFile;
+    }
+    createFolder(path) {
+        const dividedPathNames = path.split("/");
+        const folderName = dividedPathNames.pop();
+        const parentFolderPath = dividedPathNames.join("/") || "/";
```

Comment:
why again do a join, this is actually the `path` itself right?

Reply:
No Sir, This join is done after removing current file or folder name to check for its parent path.